### PR TITLE
I 708 update NSA to use the team scoreboard display format

### LINKED ISSUE #708: Update NSA to use Team Scoreboard Display Format to format team names
**Is your feature request related to a problem?** 

The NewScoringAlgorithm (NSA) does not use
the Team Scoreboard Display Format for team names.

Whereas the DSA does use the Team Scoreboard Display Format
and that is refected on the scoreboaed html and Standings tab

Note Issues #266 and #288 provided support for defining a Team Scoreboard Display Format string in both contest.yaml and via the Admin GUI

**Feature Description**: 

Add support for the Team Scoreboard Display Format for team names
in the NSA.

**Have you considered other ways to accomplish the same thing?** 

No other way.

**Do you have any specific suggestions for how your feature would be ***implemented*** in PC^2?**

Like in the DSA use the following method to format the team name

    ScoreboardVariableReplacer.substituteDisplayNameVariables(teamVarDisplayString, account, group));

**Additional context**:

### PR DESCRIPTION
**Pull request by lane55**
### Description of what the PR does

Updates NSA to use team scoreboard display format

Adds Standings XML (NSA) Report

Adds "Standings New" NSA Standings tab to scoreboard 

### Issue which the PR fixes

#708

### Environment in which the PR was developed (OS,IDE, Java version, etc.)

Microsoft Windows [Version 10.0.19044.2130]

java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)

### Precise steps for _testing_ the PR (i.e., how to demonstrate that it works correctly)

with any sample contest
start server
start admin
add board 1
start scoreboard 1

Compare Standings and Standings New tab - team names should match team names

On Admin update  team scoreboard display format
  Settings -> Team Scoreboard Display Format

On scoreboard UI team names will update to mathch format

On scoreboard UI, both Standings and Standings New tab - team names should match team names

### CODE REVIEW

File: src/edu/csus/ecs/pc2/core/scoring/NewScoringAlgorithm.java [hunk 1]
```diff
@@ -519,7 +527,10 @@ private IMemento addTeamMemento(IMemento mementoRoot, IInternalContest contest,
         standingsRecordMemento.putInteger("rank", standingsRecord.getRankNumber());
         standingsRecordMemento.putInteger("index", indexNumber);
         Account account = contest.getAccount(standingsRecord.getClientId());
-        standingsRecordMemento.putString("teamName", account.getDisplayName());
+ 
+        standingsRecordMemento.putString("teamName", ScoreboardVariableReplacer.substituteDisplayNameVariables(teamVarDisplayString, account, group));
+//        standingsRecordMemento.putString("teamName", account.getDisplayName());
```
Comment (johnbrvc) [inline src/edu/csus/ecs/pc2/core/scoring/NewScoringAlgorithm.java:532]: Should the commented out old line be removed?  it is always available from git, and it clutters the code.

Comment (lane55) [reply src/edu/csus/ecs/pc2/core/scoring/NewScoringAlgorithm.java]: removed, pushed fix.

**Review by johnbrvc — APPROVED** [review]

**Comment by troy2914** [comment]
This needs to be rebased,  and that conflicting file addressed.

File: src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+            // TODO Auto-generated method stub
+            // ignore
+        }
+
+        public void refreshRuns(RunEvent event) {
+            refreshStandings();
+        }
+
+        public void runChanged(RunEvent event) {
+            // TODO Auto-generated method stub
```
Comment (troy2914) [inline src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java:333]: I don't understand why that TODO is still listed there.

File: src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+            if (event.getAction().equals(Action.CHANGED)) {
+                refreshStandings();
+            }
+        }
+
+        public void runRemoved(RunEvent event) {
+            // TODO Auto-generated method stub
```
Comment (troy2914) [inline src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java:340]: Or here.

**Review by troy2914 — CHANGES_REQUESTED** [review]
You have a conflict right now,  please rebase.

File: src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java (hunk 2 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java:333]: ancient auto generated comment

File: src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java (hunk 3 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java:340]: ancient auto generated comment

File: src/edu/csus/ecs/pc2/ui/ReportPane.java
[the remark this answers is not in the document]
Comment (lane55) [reply src/edu/csus/ecs/pc2/ui/ReportPane.java:1]: conflict resolved and pushed

File: src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java (hunk 2 above)
Comment (troy2914) [reply src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java:333]: Not ancient,  per the diffs this is  new class.

File: src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java (hunk 3 above)
Comment (troy2914) [reply src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java:340]: Not ancient,  per the diffs this is  new class.

File: src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java (hunk 2 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java]: fixed pushed

File: src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java (hunk 3 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java]: ancient eclipse default comment

Comment (lane55) [reply src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java]: pushed fix

**Review by troy2914 — APPROVED** [review]

**Pull request merged**